src/khoj/interface/web/chat.html

```diff
@@ -869,59 +863,78 @@
+                                resolve();
+                                return;
+                            }
+                        }
 
-                let fileObj = new Blob([fileContents], { type: fileType });
-                formData.append("files", fileObj, file.name);
-                console.log(formData);
+                        let fileObj = new Blob([fileContents], { type: fileType });
+                        formData.append("files", fileObj, file.name);
+                        resolve();
+                    };
+                    reader.onerror = reject;
+                    reader.readAsArrayBuffer(file);
+                });
+            });
 
-                fetch("/api/v1/index/update?force=false&client=web", {
-                    method: "POST",
-                    body: formData,
-                })
-                    .then((data) => {
-                        console.log(data);
-                        dropzone.classList.remove('dragover');
-                        var overlayText = document.getElementById("dropzone-overlay");
-                        if (overlayText != null) {
-                            overlayText.remove();
-                        }
-                        // Display indexing success message
-                        flashStatusInChatInput("✅ File indexed successfully");
-                        renderAllFiles();
-                        addFileFilterToConversation(fileName);
-                        loadFileFiltersFromConversation();
-                    })
-                    .catch((error) => {
-                        console.log(error);
-                        dropzone.classList.remove('dragover');
-                        var overlayText = document.getElementById("dropzone-overlay");
-                        if (overlayText != null) {
-                            overlayText.remove();
-                        }
-                        // Display indexing failure message
-                        flashStatusInChatInput("⛔️ Failed to upload file for indexing");
+            // Wait for all files to be read before making the fetch request
+            Promise.all(fileReadPromises)
+                .then(() => {
+                    return fetch("/api/v1/index/update?force=false&client=web", {
+                        method: "POST",
+                        body: formData,
                     });
-            };
-
-            reader.readAsArrayBuffer(file);
+                })
+                .then((data) => {
+                    console.log(data);
+                    dropzone.classList.remove('dragover');
+                    var overlayText = document.getElementById("dropzone-overlay");
+                    if (overlayText != null) {
+                        overlayText.remove();
+                    }
+                    // Display indexing success message
+                    flashStatusInChatInput("✅ File indexed successfully");
```

Comment:
Awesome! And this should still work the same for the single file upload scenario, right?